emulators/vpcs/Portfile

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:filetype=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+
+name                vpcs
+version             0.8c
+revision            126
+categories          emulators
+license             BSD
+maintainers         {gmail.com:lasticppc @lastic} openmaintainer
+platforms           darwin
+
+description         Virtual PC Simulator built from SVN.
+long_description    ${description} The VPCS can simulate up to 9 PCs. \
+                    You can ping/traceroute them, or ping/traceroute the other hosts/routers \
+                    from the virtual PCs when you study the Cisco routers in the Dynamips. \
+                    VPCS is not the traditional PC, it is just a program running on the Linux or Windows, \
+                    and only few network commands can be used in it. \
+                    But VPCS can give you a big hand when you study the Cisco devices in the Dynamips. \
+                    VPCS can replace the routers or VMware boxes which are used as PCs in the Dynamips network.\
+                    Try VPCS, it can save your CPU/Memory. It is very small.
+
+homepage            https://wiki.freecode.com.cn/doku.php?id=wiki:vpcs
+
+master_sites        https://sourceforge.net/code-snapshots/svn/v/vp/vpcs/code
+distfiles           vpcs-code-r126-trunk.zip
+
+livecheck.type      none
+
+use_zip             yes
+
+patchfiles          patch-arch-ppc-fix.diff \
+                    patch-nonlinux-test-fix.diff
+
+fetch.type          svn
+
+svn.url		    svn://svn.code.sf.net/p/vpcs/code/trunk
```

Comment:
It is not acceptable to grab from the latest trunk version of the repository because it will change all the time. This means the buildbots will not not be in sync and also we cannot debug issues. Instead use a release version or at least a specified tag.

Reply:
The reason that I'm using the trunk (which hasn't changed since Nov 2016) is because there is a fix 0.8c which isn't in the release 0.8b.

https://sourceforge.net/p/vpcs/discussion/vpcs/thread/a1e19252/

My first time creating a port, I will have a look if I can use a specified tag instead.

Reply:
So here is where we use your revision 126. I will have to look up how that is done, I haven't fiddled with many svn ports. I usually just search the port tree for an example from someone who is skilled -- go to the top of the ports tree, and "ag svn.url . " will get some examples.

Reply:
> The reason that I'm using the trunk (which hasn't changed since Nov 2016) is because there is a fix 0.8c which isn't in the release 0.8b.
> 
> https://sourceforge.net/p/vpcs/discussion/vpcs/thread/a1e19252/

What we would usually do if there is no specific tag is use the last version 0.8b and then add a patch for the fix.

Reply:
> So here is where we use your revision 126. I will have to look up how that is done, I haven't fiddled with many svn ports. I usually just search the port tree for an example from someone who is skilled -- go to the top of the ports tree, and "ag svn.url . " will get some examples.

Out of the 20,000 ports, about 20 use the svn.url system of fetching, so there are very few examples. The best one will be by Jeremy, who basically wrote MacPorts. Check out the llvm* ports, like llvm-devel or llvm-7.0 for where he used it. 

If you don't see what you need there, there are a few others using that method.

Reply:
I thought to resolve by not using svn but then I get the curl issue.

master_sites        https://sourceforge.net/code-snapshots/svn/v/vp/vpcs/code/

distfiles           ${distname}-r126-trunk.zip

--->  Attempting to fetch vpcs-code-r126-trunk.zip from https://sourceforge.net/code-snapshots/svn/v/vp/vpcs/code/

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0DEBUG: Fetching distfile failed: error:1407742E:SSL routines:SSL23_GET_SERVER_HELLO:tlsv1 alert protocol version
--->  Attempting to fetch vpcs-code-r126-trunk.zip from http://sea.us.distfiles.macports.org/macports/distfiles/vpcs

Reply:
You are going to have to fix the curl issue if you don't want to be miserable developing ports. Just using ports, you don't notice so much, as MacPorts mirrors all the distfiles on a mirror with old security that Tiger and Leopard and other systems can still access.

But when you're writing them yourself, no mirroring.

You don't need to uninstall MacPorts to fix the curl issue. It takes no more than five minutes to build MacPorts from source as per the ticket you referenced, using the current curl already in /opt/local . You install MacPorts right on top of your existing MacPorts, no trouble at all. It replaces it.

BUT -- if you read further down that ticket -- you will see it is fragile. If you uninstall curl in MacPorts, then your MacPorts installation is BROKEN and you can't get it working again. 

Better to install another mini-MacPorts installation in /opt/bootstrap (see the instructions in that ticket) and then build your real /opt/local MacPorts against the curl in /opt/bootstrap. Then it is not fragile.

That is the price we pay for older systems wanting to use current modern SSL security. There is no way around it. Have to do something.

macports has considered bundling in a current curl version, but that idea was shot down as people don't want to be responsible for maintaining it.

Reply:
Done, and now I can get the file without the SVN trunk, yeah !

Also solves my earlier question today Ken regarding mps-youtube.